relay.go

```diff
@@ -582,13 +585,13 @@ func (r *Relayer) handleNonCallReq(f *Frame) error {
 	sent, failure := item.destination.Receive(f, frameType)
 	if !sent {
 		r.failRelayItem(items, originalID, failure, errFrameNotSent)
```

Comment:
check whether `failRelayItem` should be called in the cases that previously returned `sent=true`, and now are `sent=false`. You could rely on the `failure` being non-empty instead if you want to keep the calls behave the same (and keep using an empty error) though that is a little subtle

Reply:
with the changes in `Receive()` reverted, I don't think we have any more cases where the value of `sent` has changed, so I think we should be ok for now?